apps/docs/pages/guides/ai/going-to-prod.mdx

```diff
@@ -0,0 +1,100 @@
+}
+
+This guide will help you to prepare your application for production. We'll provide actionable steps to help you scale your application, ensure that it is reliable, can handle the load, and provide optimal precision for your use case.
+
+See our [Engineering for Scale](/docs/guides/ai/engineering-for-scale) guide for more information about engineering at scale.
+
+## Do you need indexes?
+
+Sequential scans will result in significantly higher latencies and lower throughput, guaranteeing 100% precision and not being RAM bound.
+
+There are a couple of cases where you might not need indexes:
+
+- You have a small dataset and don't need to scale it.
+- You are not expecting high amounts of vector search queries per second.
+- You need to guarantee 100% precision.
+
+You don't have to create indexes in these cases and can use sequential scans instead. This type of workload will not be RAM bound and will not require any additional resources but will result in higher latencies and lower throughput. Extra CPU cores may help to improve queries per second, but it will not help to improve latency.
+
+On the other hand, if you need to scale your application, you will need to create indexes. This will result in lower latencies and higher throughput, but will require additional RAM to make use of Postgres Caching. Also, using indexes will result in lower precision, as with indexes, you replace exact (KNN) search with approximate (ANN) search.
+
+## Understanding `probes` and `lists`
+
+Index used for approximate vector similarity search in pgvector divides a dataset into partitions. Number of these partitions is defined by the `lists` constant. The `probes` controls how many lists are going to be searched during a query.
+
+The values of lists and probes directly affect precision and RPS.
+
+- Higher `lists` means an index will be built slower, but you can achieve better RPS and precision.
+- Higher `probes` means that select queries will be slower, but you can achieve better precision.
+- `lists` and `probes` are not independent. Higher `lists` means that you will have to use higher `probes` to achieve the same precision.
+
+You can find more examples of how `lists` and `probes` constants affect precision and RPS in [pgvector 0.4.0 performance](https://supabase.com/blog/pgvector-performance) blogpost.
+
+<div>
+  <img
+    alt="multi database"
+    className="dark:hidden"
+    src="/docs/img/ai/going-prod/lists-count--light.png"
+  />
+  <img
+    alt="multi database"
+    className="hidden dark:block"
+    src="/docs/img/ai/going-prod/lists-count--dark.png"
+  />
+</div>
+
+## Performance Tips when using indexes
+
+First, a few generic tips which you can pick and choose from:
+
+1. If you are using Supabase you don't need to change your Postgres configs. But if you self-host, you should **adjust your Postgres config.** aligned with RAM & CPU cores. [Find more details here](https://gist.github.com/egor-romanov/323e2847851bbd758081511785573c08).
```

Comment:
More details [here](https://supabase.slack.com/archives/C01D6TWFFFW/p1689160970870749), but in general it may be complicated to keep up with updates.

Also I suppose that this doc will be mostly used by our platform users, and therefore they will get always up-to-date config optimizations.

Reply:
it can, but i don't want to be responsible for updating it 😞

Reply:
Got it. Perhaps something like "See example optimization"?
```markdown
1. The Supabase managed platform will automatically optimize your Postgres configs for you based on your compute addon. But if you self-host, consider **adjusting your Postgres config** based on RAM & CPU cores. See [example optimizations](https://gist.github.com/egor-romanov/323e2847851bbd758081511785573c08) for more details.
```

Reply:
love it